internal/controller/pipelinerollout_controller.go

```diff
@@ -155,35 +154,22 @@ func (r *PipelineRolloutReconciler) reconcile(
 		controllerutil.AddFinalizer(pipelineRollout, finalizerName)
 	}
 
-	// apply Pipeline
-	// todo: store hash of spec in annotation; use to compare to determine if anything needs to be updated
-	obj := kubernetes.GenericObject{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "Pipeline",
-			APIVersion: "numaflow.numaproj.io/v1alpha1",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:            pipelineRollout.Name,
-			Namespace:       pipelineRollout.Namespace,
-			OwnerReferences: []metav1.OwnerReference{*metav1.NewControllerRef(pipelineRollout.GetObjectMeta(), apiv1.PipelineRolloutGroupVersionKind)},
-		},
-		Spec: pipelineRollout.Spec.Pipeline,
+	// make a Pipeline object and add/update spec hash on the PipelineRollout object
+	obj, rolloutChildOp, err := makeChildResourceFromRolloutAndUpdateSpecHash(ctx, r.restConfig, pipelineRollout)
+	if err != nil {
+		numaLogger.Errorf(err, "failed to make a Pipeline object and to update the PipelineRollout: %v", err)
+		return false, err
 	}
 
-	// Get the object to see if it exists
-	_, err := kubernetes.GetResource(ctx, r.restConfig, &obj, "pipelines")
-	if err != nil {
-		// create object as it doesn't exist
-		if apierrors.IsNotFound(err) {
-			err = kubernetes.CreateCR(ctx, r.restConfig, &obj, "pipelines")
-			if err != nil {
-				return false, err
-			}
+	if rolloutChildOp == RolloutChildNew {
+		err = kubernetes.CreateCR(ctx, r.restConfig, obj, "pipelines")
+		if err != nil {
+			return false, err
 		}
-	} else {
+	} else if rolloutChildOp == RolloutChildUpdate {
```

Comment:
@afugazzotto @xdevxy So, the logic in here only occurs if the spec of the child object has changed, right? I think there's some logic in here related to pausing that we need to do even if that's not the case, isn't there? Such as calling `isPipelinePaused()`

Reply:
Good catch! I also noticed that we need to call `processPipelineStatus` to update the status correctly. I'm making those change now.

Reply:
I wonder if we should only set the annotation on the Rollout spec until we truly update the object itself.

Reply:
I tend to think so, right? The annotation should represent that which has actually been deployed. 

In the pause logic below we're essentially trying to update the spec. If we don't set the annotation until we've actually deployed the object to that spec, then it is okay to use the conditional "if spec is different" to determine whether to execute that logic.

Reply:
So, it may be best to set the hash annotation only after a successful call to `ApplyCRSpec`? Would it make sense to set the hash annotation together with the status (ex: in `processPipelineStatus`)?

Reply:
> So, it may be best to set the hash annotation only after a successful call to ApplyCRSpec?

yes

> Would it make sense to set the hash annotation together with the status (ex: in processPipelineStatus)?

`processPipelineStatus()` should be called all the time

Reply:
https://github.com/numaproj/numaplane/pull/56